packages/request-handler/src/docker.ts

```diff
@@ -47,44 +47,117 @@ async function getContainersWithStatsAsync() {
-      // The below type is probably wrong, sometimes stats can be undefined
-      // See https://docs.docker.com/reference/cli/docker/container/stats/ how it is / was calculated
-      memoryUsage:
-        stats.memory_stats.usage -
-        // eslint-disable-next-line @typescript-eslint/no-unnecessary-condition
-        (stats.memory_stats.stats?.cache ??
-          // eslint-disable-next-line @typescript-eslint/no-unnecessary-condition
-          stats.memory_stats.stats?.total_inactive_file ??
-          // eslint-disable-next-line @typescript-eslint/no-unnecessary-condition
-          stats.memory_stats.stats?.inactive_file ??
-          0),
-      image: container.Image,
-      ports: container.Ports,
-    };
+    try {
+      const stats = await instance.getContainer(container.Id).stats({ stream: false, "one-shot": true });
+
+      // Mark containers with invalid stats as orphaned (orphaned/errored containers)
+      // These containers often have null/undefined cpu_stats.online_cpus
+      if (!isStatsValid(stats)) {
+        return {
+          id: container.Id,
+          name: container.Names[0]?.split("/")[1] ?? "Unknown",
+          state: "orphaned" as ContainerState,
+          iconUrl:
+            dbIcons.find((icon) => {
+              const extractedImage = extractImage(container);
+              if (!extractedImage) return false;
+              return icon.name.toLowerCase().includes(extractedImage.toLowerCase());
+            })?.url ?? null,
+          cpuUsage: 0,
+          memoryUsage: 0,
+          image: container.Image,
+          ports: container.Ports,
+        };
+      }
+
+      const cpuUsage = calculateCpuUsage(stats);
+      const memoryUsage = calculateMemoryUsage(stats);
+
+      return {
+        id: container.Id,
+        name: container.Names[0]?.split("/")[1] ?? "Unknown",
+        state: container.State as ContainerState,
+        iconUrl:
+          dbIcons.find((icon) => {
+            const extractedImage = extractImage(container);
+            if (!extractedImage) return false;
+            return icon.name.toLowerCase().includes(extractedImage.toLowerCase());
+          })?.url ?? null,
+        cpuUsage,
+        memoryUsage,
+        image: container.Image,
+        ports: container.Ports,
+      };
+    } catch (error) {
+      // Mark containers that fail to fetch stats as orphaned
+      return {
+        id: container.Id,
+        name: container.Names[0]?.split("/")[1] ?? "Unknown",
+        state: "orphaned" as ContainerState,
+        iconUrl:
+          dbIcons.find((icon) => {
+            const extractedImage = extractImage(container);
+            if (!extractedImage) return false;
+            return icon.name.toLowerCase().includes(extractedImage.toLowerCase());
+          })?.url ?? null,
+        cpuUsage: 0,
+        memoryUsage: 0,
+        image: container.Image,
+        ports: container.Ports,
+      };
```

Comment:
Do we really want to show "failed containers" as "orphaned"?
When does this catch usually get triggered?